src/components/EzField/EzChoice.tsx

```diff
@@ -46,25 +47,43 @@ const box = theme.css({
   ],
 });
 
+const nestedContent = theme.css({
+  marginLeft: '$300',
+});
+
 const inputStyles = theme.css({marginRight: '$100'});
 
-const Option = ({input, bordered, disabled, label}) => {
+const Option = ({input, bordered, disabled, label, labelFromChildren, contentFromChildren}) => {
   const id = useUniqueId();
+  const childLabelWithHtmlFor = React.cloneElement(labelFromChildren, {htmlFor: id});
   return (
     <Style ruleset={theme}>
       <span className={box({bordered, disabled})}>
         {React.cloneElement(input, {id})}
-        <label htmlFor={id}>{label}</label>
+        {(labelFromChildren && childLabelWithHtmlFor) || <label htmlFor={id}>{label}</label>}
```

Comment:
```suggestion
        {(childLabelWithHtmlFor) || <label htmlFor={id}>{label}</label>}
```

Presumably if `labelFromChildren` is `undefined`, `childLabelWithHtmlFor` should also be `undefined`? Looking at this a little further, if `labelFromChildren` is `undefined` (or `null`), I would have expected the `React.cloneElement` call to fail. 

What am I missing?

Reply:
I think this code relies on `labelFromChildren` being `false` (rather than `undefined` or `null`), otherwise `React.cloneElement` would throw: 

![image](https://user-images.githubusercontent.com/109814/126340253-a043a6fe-fc77-4e09-82a0-66dd55d16ad9.png)

Passing `false` for `labelFromChildren` doesn't seem like it should be valid (if we had type defs for the arguments instead of `any`, the compiler would be mad about this).

Reply:
FWIW, if we update `EzLabel` to use [`Slot`](https://github.com/ezcater/recipe/blob/main/src/components/EzContent/Slot.tsx), you could skip using `React.cloneElement` and use [SlotProvider](https://github.com/ezcater/recipe/blob/main/src/utils/slots.ts#L21-L41) to wire up the `htmlFor` attribute:

```diff
const EzLabel = forwardRef<Ref, Props>(({as: element, ...initialProps}, ref) => {
  const {props} = styles(initialProps);

  return (
    <Style ruleset={theme}>
--      <Element {...props} ref={ref} />
++      <Slot element={element} slot="label" {...props} ref={ref} /> 
    </Style>
  );
});

```
We can also use the `SlotProvider` to wire up the `className` for the content (rather than wrapping with an additional `div`):

```diff
const Option = ({input, bordered, disabled, label, labelFromChildren, contentFromChildren}) => {
  const id = useUniqueId();
  return (
++  <SlotProvider slots={{label: {htmlFor: id}, content: {className: nestedContent()}}}>
      <Style ruleset={theme}>
        <span className={box({bordered, disabled})}>
          {React.cloneElement(input, {id})}
--        {(labelFromChildren && childLabelWithHtmlFor) || <label htmlFor={id}>{label}</label>}
++        {(labelFromChildren) || <Slot element="label" slot="label">{label}</Slot>}
        </span>
--      {contentFromChildren && <div className={nestedContent()}>{contentFromChildren}</div>}
++      {contentFromChildren}
      </Style>
++  </SlotProvider>
  );
};
```